### PR TITLE
fix: Broken backup handling of empty topics

### PR DESCRIPTION
**Pull request by aiven-anton**
### About this change - What it does

- Remove an incorrect assertion on number of data files in `Metadata.__post_init__`. Empty partitions will not have data files.
- Add integration test verifying a full backup-restoration cycle for an empty partition.

### CODE REVIEW

File: tests/integration/backup/test_v3_backup.py [hunk 1]
```diff
@@ -181,7 +190,10 @@ def test_roundtrip_from_kafka_state(
         check=True,
     )
 
-    # Verify restored topic.
+    # Verify configuration is identical.
+    assert topic_config == get_topic_configurations(admin_client, new_topic.name, {ConfigSource.TOPIC_CONFIG})
```
Comment (aiven-anton) [inline tests/integration/backup/test_v3_backup.py:194]: I moved this so that it doesn't happen in-between reading and verifying of records.
